packages/drawer/src/Drawer.tsx

```diff
@@ -70,12 +72,39 @@ export default function Drawer({
   open = false,
   showCloseButton = false,
 }: DrawerProps) {
+  const drawerHiddenStateMargin = -1000
   const [isOpen, setIsOpen] = useState(false)
+  const [drawerHiddenMargin, setDrawerHiddenMargin] = useState(drawerHiddenStateMargin)
+  const [backdropOpacity, setBackdropOpacity] = useState(0)
+
+  const playShowDrawerAnimation = () => {
+    // We want to execute it after the rerender
+    setTimeout(() => {
+      setDrawerHiddenMargin(0)
+      setBackdropOpacity(1)
+    }, 1)
+  }
+
+  const playHideDrawerAnimation = () => {
+    // We want to execute it after the rerender
+    setTimeout(() => {
+      setBackdropOpacity(0)
+      setDrawerHiddenMargin(drawerHiddenStateMargin)
+    }, 1)
```

Comment:
Remove `1` parameter

Reply:
If I remove it or set to 0, it actually isn't getting executed on the next rerender and instead of animation the backdrop just blinks into the screen. Steps to reproduce:

1. Remove 1 parameter
2. Open the storybook locally
3. Refresh the Drawer story and open a different drawer story (if you don't refresh and do story switching I think something is getting cached and it works properly)
4. Animation won't work properly on the first drawer opening (while it works okay when I open the drawer for the second time)